internal/pkg/deploy/cloudformation/stack/task.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack
+
+import (
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"strconv"
+
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+const (
+	taskTemplatePath = "task/cf.yml"
+
+	taskNameParamKey         = "TaskName"
+	taskCPUParamKey          = "TaskCPU"
+	taskMemoryParamKey       = "TaskMemory"
+	taskLogRetentionParamKey = "LogRetention"
+
+	taskContainerImageParamKey = "ContainerImage"
+	taskTaskRoleParamKey       = "TaskRole"
+	taskCommandParamKey        = "Command"
+
+	taskLogRetentionInDays = "1"
+)
+
+type taskStackConfig struct {
+	*deploy.CreateTaskResourcesInput
+	parser template.ReadParser
+}
+
+// NewTaskStackConfig sets up a struct that provides stack configurations for CloudFormation
+// to deploy the task resources stack.
+func NewTaskStackConfig(taskOpts *deploy.CreateTaskResourcesInput) *taskStackConfig {
+	return &taskStackConfig{
+		CreateTaskResourcesInput: taskOpts,
+		parser: template.New(),
+	}
+}
+
+// StackName returns the name of the CloudFormation stack for the task.
+func (t *taskStackConfig) StackName() string {
+	return NameForTask(t.Name)
+}
+
+// Template returns the task CloudFormation template.
+func (t *taskStackConfig) Template() (string, error) {
+	content, err := t.parser.Parse(taskTemplatePath, struct{
+		EnvVars map[string]string
+	}{
+		EnvVars: t.EnvVars,
+	})
+	if err != nil {
+		return "", fmt.Errorf("read template for task stack: %w", err)
+	}
+	return content.String(), nil
+}
+
+// Parameters returns the parameter values to be passed to the task CloudFormation template.
+func (t *taskStackConfig) Parameters() ([]*cloudformation.Parameter, error) {
+	return []*cloudformation.Parameter{
+		{
+			ParameterKey:   aws.String(taskNameParamKey),
+			ParameterValue: aws.String(t.Name),
+		},
+		{
+			ParameterKey:   aws.String(taskCPUParamKey),
+			ParameterValue: aws.String(strconv.Itoa(t.CPU)),
+		},
+		{
+			ParameterKey:   aws.String(taskMemoryParamKey),
+			ParameterValue: aws.String(strconv.Itoa(t.Memory)),
+		},
+		{
+			ParameterKey:   aws.String(taskLogRetentionParamKey),
+			ParameterValue: aws.String(taskLogRetentionInDays),
+		},
+		{
+			ParameterKey:   aws.String(taskContainerImageParamKey),
+			ParameterValue: aws.String(t.Image),
+		},
+		{
+			ParameterKey:   aws.String(taskTaskRoleParamKey),
+			ParameterValue: aws.String(t.TaskRole),
+		},
+		{
+			ParameterKey:   aws.String(taskCommandParamKey),
+			ParameterValue: aws.String(t.Command),
+		},
+	}, nil
+}
+
+// Tags returns the tags that should be applied to the task CloudFormation.
+func (t *taskStackConfig) Tags() []*cloudformation.Tag {
+	appEnvTags := make(map[string]string)
+
+	if t.Env != config.EnvNameNone {
```

Comment:
yeah this is a bit strange that the `stack` pkg should have a dependency on `config`, I think in the next PR I'd love for us to remove `config.EnvNameNone` and just have the default value `""` mean that an environment was not set.